### ISSUE TITLE
Internet access is required for running azure functions on local machine

### ISSUE THREAD

**Issue by William1104** [Azure/azure-functions-core-tools#3821]
### Version

```
$ func --version
4.0.6280
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
```

### Description

`func start` would fail to execute when there is no Internet access. Some developers may have to work in a network without Internet access (or very limited access). It potentially makes azure-functions-core-tools not work..

When the machine has Internet access, we can run the tools locally.
```
$ func start

Azure Functions Core Tools
Core Tools Version:       4.0.6280 Commit hash: N/A +421f0144b42047aa289ce691dc6db4fc8b6143e6 (64-bit)
Function Runtime Version: 4.834.3.22875

[2024-09-12T16:11:30.915Z] OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.

Functions:

        echo: [GET,POST] http://localhost:7071/api/echo

        hello: [GET,POST] http://localhost:7071/api/hello

        uppercase: [GET,POST] http://localhost:7071/api/uppercase

For detailed output, run func with --verbose flag.
[2024-09-12T16:11:31.526Z] Worker process started and initialized.
[2024-09-12T16:11:35.788Z] Host lock lease acquired by instance ID '000000000000000000000000E7A1833D'.
^C[2024-09-12T16:11:41.082Z] Language Worker Process exited. Pid=6500.
[2024-09-12T16:11:41.082Z] /usr/lib/jvm/java-21-openjdk-amd64/bin/java exited with code 130 (0x82). .
```

But when we stop the Internet access, the tool would fail to build configuration in an external class. It looks like this process needs to download something from a CDN (functionscdn.azureedge.net:443). 
```
$ func start

Azure Functions Core Tools
Core Tools Version:       4.0.6280 Commit hash: N/A +421f0144b42047aa289ce691dc6db4fc8b6143e6 (64-bit)
Function Runtime Version: 4.834.3.22875

[2024-09-12T16:12:09.804Z] Error building configuration in an external startup class.
[2024-09-12T16:12:09.804Z] Error building configuration in an external startup class. System.Net.Http: Resource temporarily unavailable (functionscdn.azureedge.net:443). System.Net.Sockets: Resource temporarily unavailable.
[2024-09-12T16:12:09.806Z] A host error has occurred during startup operation 'b37baf23-c6f0-476d-a961-41fd6ebcdbf5'.
[2024-09-12T16:12:09.806Z] Microsoft.Azure.WebJobs.Script: Error building configuration in an external startup class. System.Net.Http: Resource temporarily unavailable (functionscdn.azureedge.net:443). System.Net.Sockets: Resource temporarily unavailable.
Value cannot be null. (Parameter 'provider')
[2024-09-12T16:12:09.823Z] Host startup operation has been canceled
```

### Steps to reproduce

simply stop the Internet access and then run `func start`. We should see error following messages:

```
[2024-09-12T16:12:09.804Z] Error building configuration in an external startup class.
[2024-09-12T16:12:09.804Z] Error building configuration in an external startup class. System.Net.Http: Resource temporarily unavailable (functionscdn.azureedge.net:443). System.Net.Sockets: Resource temporarily unavailable.
[2024-09-12T16:12:09.806Z] A host error has occurred during startup operation 'b37baf23-c6f0-476d-a961-41fd6ebcdbf5'.
[2024-09-12T16:12:09.806Z] Microsoft.Azure.WebJobs.Script: Error building configuration in an external startup class. System.Net.Http: Resource temporarily unavailable (functionscdn.azureedge.net:443). System.Net.Sockets: Resource temporarily unavailable.
```

**Comment by kshyju** [comment]
@umangsriv Could you investigate this? You can use fiddler to find out what exact URL it is pinging.

**Comment by umangsriv** [comment]
> @umangsriv Could you investigate this? You can use fiddler to find out what exact URL it is pinging.
> 
> May be it's related to [Azure/azure-functions-host#9434](https://github.com/Azure/azure-functions-host/issues/9434)

Sure @kshyju , I'll look into this.

**Comment by umangsriv** [comment]
Hey @kshyju, could you please provide an update so we can proceed?

**Comment by liliankasem** [comment]
@kshyju just a heads up that there are updates here! Thanks :)

**Comment by satishH2N** [comment]
as of today my azure function is not even starting can any help please this started happen suddenly  

logs:-

[2025-08-03T07:38:24.478Z] Error building configuration in an external startup class. System.Net.Http: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing. The operation was canceled. System.Net.Http: The operation was canceled. System.Net.Sockets: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request.. The I/O operation has been aborted because of either a thread exit or an application request.
[2025-08-03T07:38:29.495Z] A host error has occurred during startup operation '2b946edd-55d9-4e2c-bcf8-6e4faaf5063a'.
[2025-08-03T07:38:29.497Z] Microsoft.Azure.WebJobs.Script: Error building configuration in an external startup class. System.Net.Http: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing. The operation was canceled. System.Net.Http: The operation was canceled. System.Net.Sockets: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request.. The I/O operation has been aborted because of either a thread exit or an application request.  
Value cannot be null. (Parameter 'provider')
[2025-08-03T07:38:29.552Z] Host startup operation has been canceled


A host error has occurred during startup operation 'bde7f3f0-4f77-4175-ae67-82df4641f492'.
[2025-08-03T08:24:04.566Z] Microsoft.Azure.WebJobs.Script: Error building configuration in an external startup class. System.Net.Http: No such host is known. (cdn.functions.azure.com:443). System.Net.Sockets: No such host is known.
Value cannot be null. (Parameter 'provider')

**Comment by sandroasp** [comment]
With this limitation, we cannot open the Logic App designer offline. This is a huge limitation! It doesn't allow developers to develop simple request/response workflows that the only thing that it does is call a XSLT map, Neither troubleshoot an existing workflow in terms of development the implementation since it doesn't open the workflow

<img width="1291" height="427" alt="Image" src="https://github.com/user-attachments/assets/f6b6c26e-40e0-4187-868d-4e75d9048424" />

Do we have an idea if this issue is going to be fixed?

**Comment by pedropcamellon** [comment]
**Workaround: Commenting out `extensionBundle` resolves local startup**

Commenting out the `extensionBundle` configuration in `host.json` allows `func start` to run successfully when `functionscdn.azureedge.net` is unreachable:

```json
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      }
    }
  },
  // "extensionBundle": {
  //   "id": "Microsoft.Azure.Functions.ExtensionBundle",
  //   "version": "[4.*, 5.0.0)"
  // },
  "extensions": {
    "blobs": {
      "maxDegreeOfParallelism": 4
    }
  }
}
```

**Environment:**
- Function was working this morning (October 29, 2025)
- Suddenly stopped working around midday
- Both `ping` and `curl`/`Invoke-WebRequest` to `functionscdn.azureedge.net` timeout with no response

**Note:** This is a temporary workaround for local development only. The `extensionBundle` should remain enabled for production deployments, as it's required for binding extensions to function properly in Azure. This suggests the CDN connectivity issue is blocking extension bundle downloads during local startup.

Related to #4260  regarding CDN migration from `functionscdn.azureedge.net` to `cdn.functions.azure.com`.

**Comment by nk54** [comment]
I didn't know it needed to reach cdn.functions.azure.com while working offline.
I couldn't work when the Azure Portal stopped working this afternoon

**Comment by ldibanyez** [comment]
> I didn't know it needed to reach cdn.functions.azure.com while working offline. I couldn't work when the Azure Portal stopped working this afternoon

Yup, No CDN no party. Also learned the hard way. They should really allow caching to at least unblock local dev. (Though if you need a service that requires network infra, you are probably blocked anyway)

"Error building configuration in an external startup class. System.Net.Http: No such host is known. (cdn.functions.azure.com:443). System.Net.Sockets: No such host is known."

**Comment by liliankasem** [comment]
x-posting my response from #4696 

https://github.com/Azure/azure-functions-core-tools/issues/4696#issuecomment-3463366358

This is super unfortunate and highlights the need for this work item to be higher priority for us.

**Comment by liliankasem** [comment]
Issue should be fixed as part of the next core tools release (v4.8.0, eta for release is end of this week) - please open a new issue if you're on 4.8.0 and experiencing problems with this new feature - thanks!
